domains/authorization/Authorization.API/API/Authorization/Controllers/AdminController.cs

```diff
@@ -0,0 +1,38 @@
+using System.Threading.Tasks;
+using API.Authorization._Features_;
+using API.ValueObjects;
+using Asp.Versioning;
+using EnergyOrigin.TokenValidation.b2c;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace API.Authorization.Controllers;
+
+[ApiController]
+[Authorize(Policy = Policy.B2CCustomPolicyClientPolicy)]
+[ApiVersion(ApiVersions.Version20230101)]
+public class AdminController : ControllerBase
+{
+    private readonly IMediator _mediator;
+
+    public AdminController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    /// <summary>
+    /// Create Client.
+    /// </summary>
+    [HttpPost]
+    [Authorize(Policy.B2CCustomPolicyClientPolicy)]
```

Comment:
What does the policy enforce? That i should be an authenticated and authorized client, before i can create a client?

Reply:
This policy allows only Self / Master Issuer. The same policy used by CustomPolicy to get permissions. So not normal tokens can call.